librz/bin/format/elf/elf.c

```diff
@@ -134,36 +134,41 @@ static HtUP *rel_cache_new(RzBinElfReloc *relocs, ut32 reloc_num) {
 	return rel_cache;
 }
 
-static void set_phdr_entry(ELFOBJ *bin, size_t phdr_entry_index, ut8 *entry) {
-	size_t pos = 0;
-
-	bin->phdr[phdr_entry_index].p_type = READ32(entry, pos);
+static bool set_phdr_entry(ELFOBJ *bin, Elf_(Phdr) * segment, ut64 offset) {
 #if RZ_BIN_ELF64
-	bin->phdr[phdr_entry_index].p_flags = READ32(entry, pos);
-#endif
-	bin->phdr[phdr_entry_index].p_offset = RZ_BIN_ELF_READWORD(entry, pos);
-	bin->phdr[phdr_entry_index].p_vaddr = RZ_BIN_ELF_READWORD(entry, pos);
-	bin->phdr[phdr_entry_index].p_paddr = RZ_BIN_ELF_READWORD(entry, pos);
-	bin->phdr[phdr_entry_index].p_filesz = RZ_BIN_ELF_READWORD(entry, pos);
-	bin->phdr[phdr_entry_index].p_memsz = RZ_BIN_ELF_READWORD(entry, pos);
-#ifndef RZ_BIN_ELF64
-	bin->phdr[phdr_entry_index].p_flags = READ32(entry, pos);
+	if (!Elf_(rz_bin_elf_read_word)(bin, &offset, &segment->p_type) ||
+		!Elf_(rz_bin_elf_read_word)(bin, &offset, &segment->p_flags) ||
```

Comment:
Some field are not in the same order. (p_flags)

Reply:
Ok, good point!

Reply:
Still, I prefer the previous "format" where the common operations were outside of the ifdef and the special cases were in the ifdefs. Otherwise it is hard to compare the two things.